core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1695,6 +1701,21 @@ private Optional<ConnectorMaterializedViewDefinition> getMaterializedViewInterna
         return Optional.empty();
     }
 
+    @Override
+    public Map<String, Object> getMaterializedViewProperties(Session session, QualifiedObjectName viewName, MaterializedViewDefinition materializedViewDefinition)
+    {
+        Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, viewName.getCatalogName());
+        if (catalog.isPresent()) {
+            CatalogMetadata catalogMetadata = catalog.get();
+            CatalogHandle catalogHandle = catalogMetadata.getCatalogHandle(session, viewName);
+            ConnectorMetadata metadata = catalogMetadata.getMetadataFor(session, catalogHandle);
+
+            ConnectorSession connectorSession = session.toConnectorSession(catalogHandle);
+            return metadata.getMaterializedViewProperties(connectorSession, viewName.asSchemaTableName(), materializedViewDefinition.toConnectorMaterializedViewDefinition());
```

Comment:
Verify connector-provided value isn't null. Also important to decide whether the map may contain null values.
I think it should not (https://github.com/trinodb/trino/blob/f8e0c7cc72dc2e7bc961d7ee17ea3df5fb766d18/core/trino-main/src/main/java/io/trino/metadata/MaterializedViewDefinition.java#L55), so pass the return value thru `ImmutableMap.copyOf`